pkg/acsengine/engine.go

```diff
@@ -487,12 +487,16 @@ func getParameters(cs *api.ContainerService, isClassicMode bool) (map[string]int
 	return parametersMap, nil
 }
 
+func AddValue(m map[string]interface{}, k string, v interface{}) { addValue(m, k, v) }
 func addValue(m map[string]interface{}, k string, v interface{}) {
 	m[k] = map[string]interface{}{
 		"value": v,
 	}
 }
 
+func AddSecret(m map[string]interface{}, k string, v interface{}, encode bool) {
```

Comment:
Again, another intentional design decision. We needed this exported so we could use it in the new interpolator packages. At the end of the day it's a convenience function, and I am happy to copy this into the other packages as necessary.